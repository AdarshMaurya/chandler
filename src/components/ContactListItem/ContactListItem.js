import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {View, Text, TouchableOpacity} from 'react-native';
import {getLastUpdatedDate} from 'utils/contacts';
import {ContactAvatar} from 'components/ContactAvatar';

import {styles} from './ContactListItem.styles';
import {commonStyles} from 'theme';

export class ContactListItem extends PureComponent {
  static propTypes = {
    contact: PropTypes.object.isRequired,
    onPress: PropTypes.func.isRequired,
    bgOddRow: PropTypes.bool,
  };
  render() {
    const {contact, onPress, bgOddRow} = this.props;

    return (
      <TouchableOpacity onPress={onPress}>
        <View style={[styles.container, bgOddRow && commonStyles.bgOddRow]}>
          <ContactAvatar contact={contact} size={50} />
          <View style={styles.textContainer}>
            <Text>
              {contact.first_name} {contact.last_name}
            </Text>
            <Text style={commonStyles.mutedText}>
              Last updated {getLastUpdatedDate(contact)}
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}
