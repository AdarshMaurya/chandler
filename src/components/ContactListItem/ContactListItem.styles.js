import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: 10,
    paddingVertical: 3,
  },
  image: {
    borderRadius: 3,
    height: 50,
    width: 50,
  },
  textContainer: {
    justifyContent: 'center',
    paddingHorizontal: 10,
  },
});
