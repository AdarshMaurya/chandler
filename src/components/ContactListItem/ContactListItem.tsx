import React, {PureComponent} from 'react';
import {Text, TouchableOpacity, View} from 'react-native';

import {ContactAvatar} from '@components';
import {I18n} from '@i18n';
import {IContact} from '@models';
import {commonStyles} from '@theme';
import {getLastUpdatedDate} from '@utils/contacts';
import {styles} from './ContactListItem.styles';

interface IContactListItemProps {
  contact: IContact;
  onPress: (...args) => any;
  bgOddRow?: boolean;
}

export class ContactListItem extends PureComponent<IContactListItemProps> {
  public render() {
    const {contact, onPress, bgOddRow} = this.props;

    return (
      <TouchableOpacity onPress={onPress}>
        <View style={[styles.container, bgOddRow && commonStyles.bgOddRow]}>
          <ContactAvatar contact={contact} size={50} />
          <View style={styles.textContainer}>
            <Text>
              {contact.first_name} {contact.last_name}
            </Text>
            <Text style={commonStyles.mutedText}>
              {I18n.t('common:lastUpdated', {
                date: getLastUpdatedDate(contact),
              })}
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}
