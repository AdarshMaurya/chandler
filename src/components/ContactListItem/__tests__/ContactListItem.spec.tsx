import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {ContactListItem} from '../ContactListItem';

jest.mock('@utils/contacts', () => ({
  getLastUpdatedDate: () => 'testDate',
}));

describe('Components', () => {
  describe('ContactListItem', () => {
    it('should renders correctly', () => {
      const onPress1 = jest.fn();
      const onPress2 = jest.fn();
      expect(
        toJson(
          shallow(<ContactListItem contact={{} as any} onPress={onPress1} />)
        )
      ).toMatchSnapshot();

      expect(
        toJson(
          shallow(
            <ContactListItem
              contact={{} as any}
              onPress={onPress2}
              bgOddRow={true}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should call onPress when taping the list row', () => {
      const onPress = jest.fn();
      const button = shallow(
        <ContactListItem contact={{} as any} onPress={onPress} />
      );
      button.prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
