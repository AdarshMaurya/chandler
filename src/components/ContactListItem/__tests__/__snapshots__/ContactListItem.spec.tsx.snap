// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContactListItem should renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 3,
        },
        undefined,
      ]
    }
  >
    <ContactAvatar
      contact={Object {}}
      size={50}
    />
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
          }
        }
      >
        Last updated on testDate
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Components ContactListItem should renders correctly 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 3,
        },
        Object {
          "backgroundColor": "#FBFBFB",
        },
      ]
    }
  >
    <ContactAvatar
      contact={Object {}}
      size={50}
    />
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
          }
        }
      >
        Last updated on testDate
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;
